chia/server/ws_connection.py

```diff
@@ -223,6 +223,9 @@ async def perform_handshake(
             if inbound_handshake.network_id != network_id:
                 raise ProtocolError(Err.INCOMPATIBLE_NETWORK_ID)
 
+            if inbound_handshake.protocol_version != protocol_version:
+                raise ProtocolError(Err.INCOMPATIBLE_PROTOCOL_VERSION)
```

Comment:
both the full node- and wallet protocols need to be backwards compatible, so I don't think it's safe to be this strict for all protocols.

For the full node- and wallet protocols, we use the version to know whether new messages are supported or not.

Reply:
OK understood. Is it fine if we enforce this then only for the Harvester and Farmer protocols?